tsconfig.json

```diff
@@ -1,9 +1,9 @@
 {
   "compilerOptions": {
     "module": "commonjs",
-    "target": "ES2022",
+    "target": "ES2023",
     "outDir": "out",
-    "lib": ["ES2022", "dom", "dom.iterable", "webworker"],
+    "lib": ["ES2023", "dom", "dom.iterable", "webworker"],
     "sourceMap": true,
     "esModuleInterop": true, // added by tnaum to help with Denque integration, happy to discuss other approaches
     "allowSyntheticDefaultImports": true, // added by tnaum to fix SlickGrid integration, eager to discuss other approaches
```

Comment:
@tnaum-ms maybe unrelated to this PR but quickly browsing to this and saw that `allowSyntheticDefaultImports` was tagged as necessary for SlickGrid, but why? It seems that I also had this config in my Slickgrid-React setup and removing it in this new [PR](https://github.com/ghiscoding/slickgrid-react/pull/498) causes no issues for the 900 Cypress E2E tests that I have, it's all working fine on my end without this enabled. The only thing that I changed on the project a few months ago was to switch from WebPack to Vite. Could that be a WebPack related issue?